### ISSUE TITLE
Could this replace nixos containers?

### ISSUE THREAD

**Issue by VanCoding** [nix-community/NixNG#47]
Hi there

I'm curious: Could I run this on top of NixOS the same way NixOS-Containers work?

My thought process is the following: When I have a lot of different services running on a single machine, and each one has some child-services like a database, for example, it's cool to sandbox these systems grouped together.

NixOS Containers do just that, but they are probably a bit heavyweight. If the goal of NixNG is leightweight "containers", couldn't it also make lightweight NixOS containers. They would work exactly like NixOS containers, but with the modules&options of NixNG inside the container instead of the NixOS ones.

Do you think that would even make a big difference?

**Comment by MagicRB** [comment]
Yeah so uh that exists, https://git.sr.ht/~magic_rb/uk3s.nix/tree/master/item/nixos/modules/ucontainers/default.nix but to be honest the main problem is eval time, nixng helps a bit but still nix is slow just, well, because irs slow

**Comment by VanCoding** [comment]
Ah, interesting, I'll have a look at that.

What's the problem with eval time, though? If the services don't change that often, does it even matter? I mean, in a highly dynamic system like kubernetes, where containers change all the time, it might be an issue, but then, I'd probably use imperative containers, no?

**Comment by MagicRB** [comment]
Ah ya if you do imperative its fine. The thing i linked doesnt support imperative yet though. I do have a plan for a hybrid imperative/declarative approach similar to how i deploy into kubernetes. I just havent written it

**Comment by VanCoding** [comment]
Nice! Thanks for commenting on the subject a bit, that's all I could've asked for :)

**Comment by MagicRB** [comment]
Not completed tho :) moving ucontainers into its own repo would be nice and also docs, so ill keep this open.

**Comment by evangineer** [comment]
Would love to know about how you deploy into Kubernetes.

Do you have a link for that?

**Comment by MagicRB** [comment]
and merged.
